File: app/src/main/java/com/example/movieguide/app/adapter/OnboardingAdapter.java
package com.example.movieguide.app.adapter;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import com.example.movieguide.app.fragment.Onboarding1Fragment;
import com.example.movieguide.app.fragment.Onboarding2Fragment;
import com.example.movieguide.app.fragment.Onboarding3Fragment;

public class OnboardingAdapter extends FragmentPagerAdapter {
    private static int ITEM_COUNT = 3;

    public OnboardingAdapter(FragmentManager fragmentManager) {
        super(fragmentManager);
    }

    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case 0:
                return new Onboarding1Fragment();
            case 1:
                return new Onboarding2Fragment();
            case 2:
                return new Onboarding3Fragment();
            default:
                return null;
        }
    }

    @Override
    public int getCount() {
        return ITEM_COUNT;
    }
}

File: app/src/main/java/com/example/movie/adapter/OnboardingAdapter.kt
package com.example.movie.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import com.example.movie.model.OnboardingModel
import com.example.movie.view.IntroFragment


class OnboardingAdapter(fm: FragmentManager, list: ArrayList<OnboardingModel>): FragmentPagerAdapter(fm) {
    private var list = ArrayList<OnboardingModel>()

    init {
        this.list = list
    }

    override fun getItem(i: Int): Fragment {
        return IntroFragment.newInstance(list.get(i))
    }

    override fun getCount(): Int {
        return list.size
    }
}